src/main/java/org/folio/service/pieces/PieceUpdateInventoryService.java

```diff
@@ -63,13 +64,14 @@ public Future<String> manualPieceFlowCreateItemRecord(Piece piece, CompositePoLi
 
   public Future<Pair<String, String>> deleteHoldingConnectedToPiece(Piece piece, RequestContext requestContext) {
     if (piece != null && piece.getHoldingId() != null) {
+      var locationContext = createContextWithNewTenantId(requestContext, piece.getReceivingTenantId());
       String holdingId = piece.getHoldingId();
-      return inventoryHoldingManager.getHoldingById(holdingId, true, requestContext)
+      return inventoryHoldingManager.getHoldingById(holdingId, true, locationContext)
                   .compose(holding -> {
```

Comment:
I suggest we extract this block to a private method or perform smaller extraction work for each nested callback, if possible

Reply:
maybe lets do this refactoring in subsequent PRs because this PR already contain lot of refactoring that slows down code review and we need to finish it and merge

Reply:
@SerhiiNosko As it is not quire readable, I can modify the two places Bob mentioned without much work and take care of the rest if there will be anything needed in the future

Reply:
okay got it, let's do